protocol/nats_jetstream/v2/receiver.go

```diff
@@ -54,12 +69,20 @@ func (r *Receiver) Receive(ctx context.Context) (binding.Message, error) {
 type Consumer struct {
```

Comment:
Given that we've exposed this as a struct with named v1 fields, it feels like maybe introducing a new `v3` version of `nats_jetstream` might be the way to go.  I don't have a really strong feeling about it, though.